### PR TITLE
Make User Profile Storage models use circe instead of spray-json

### PR DESCRIPTION
**Pull request by jcazevedo**
This PR makes the User Profile Storage models use circe instead of spray-json.

### CODE REVIEW

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala [hunk 1]
```diff
@@ -26,16 +26,14 @@ case object MergeMergingStrategy extends AttributesMergingStrategy
 case object ReplaceMergingStrategy extends AttributesMergingStrategy
 
 object AttributesMergingStrategy {
-  implicit object AttributesMergingStrategyJsonFormat extends JsonFormat[AttributesMergingStrategy] {
-    def read(json: JsValue) = json match {
-      case JsString("merge") => MergeMergingStrategy
-      case JsString("replace") => ReplaceMergingStrategy
-      case strategy => deserializationError(s"Invalid attribute merging strategy: ${strategy.compactPrint}")
-    }
+  implicit val jsonEncoder: Encoder[AttributesMergingStrategy] = Encoder[String].contramap({
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala]: What do you think of leaving this with a name like `mergingStrategyJsonEncoder`? I had problems before with importing objects containing implicits with the same name, as they stop working in that situation. We shouldn't ever need to import this just for the implicits, but we may import this if we ever add other settings or classes inside this object.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala [hunk 2]
```diff
@@ -35,81 +35,79 @@ case object Older extends BasicSingleAttributeMergingStrategy
...
+ * Strings with value `"true"` are converted to booleans with value `true`.
+ * Strings with value `"false"` are converted to booleans with value `false`.
+ * Strings whose value can be parsed to a Double follow the rules of the number conversion enumerated above.
  */
 case object Or extends BasicSingleAttributeMergingStrategy
 
-object SingleAttributeMergingStrategy {
-  private val strToBasicMergeStrategy = Map[String, BasicSingleAttributeMergingStrategy](
+object BasicSingleAttributeMergingStrategy {
+  val fromStr = Map[String, BasicSingleAttributeMergingStrategy](
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala]: Can we leave this private?

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala [hunk 3]
```diff
@@ -119,8 +117,20 @@ object SingleAttributeMergingStrategy {
     "map-merge" -> MapMerge,
     "and" -> And,
     "or" -> Or)
-  private val basicMergeStrategyToStr = strToBasicMergeStrategy.map { case (k, v) => v -> k }
+  val toStr = fromStr.map { case (k, v) => v -> k }
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala]: Can we leave this private?

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala [hunk 4]
```diff
@@ -133,22 +143,32 @@ object SingleAttributeMergingStrategy {
...
+    Encoder[String].contramap {
       case PeriodicSingleAttributeMergingStrategy(ps, ms) =>
-        s"${basicMergeStrategyToStr(ms)}|${ps.length}${timeUnitToStr(ps.unit)}".toJson
+        s"${BasicSingleAttributeMergingStrategy.toStr(ms)}|${ps.length}${timeUnitToStr(ps.unit)}"
     }
-  }
+
+  implicit val jsonDecoder: Decoder[PeriodicSingleAttributeMergingStrategy] =
+    Decoder[String].emap({
+      case periodicSingleAttributeStrategyR(str, length, unit) if BasicSingleAttributeMergingStrategy.fromStr.contains(str) && strToTimeUnit.contains(unit) =>
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala]: Can we add different error messages for different failed aspects of a `PeriodicSingleAttributeMergingStrategy`?

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala (hunk 1 above)
Comment (jcazevedo) [reply sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala]: Makes sense. I've renamed all Encoders and Decoders accordingly in 
8542515.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala (hunk 3 above)
Comment (jcazevedo) [reply sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala]: Yes. Changed it in d7f5222.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala (hunk 2 above)
Comment (jcazevedo) [reply sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala]: Yes. Changed it in d7f5222.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala (hunk 4 above)
Comment (jcazevedo) [reply sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala]: Yes. I ended up relying on apso's `Decoder` for `FiniteDuration`. This has the disadvantage of not being able to parse durations in microseconds, but I'm seriously doubting we need microseconds for a `PeriodicSingleAttributeMergingStrategy`.

**Pull request merged**